### PR TITLE
Ternary Search in Ruby (#2176)

### LINKED ISSUE #2176: Ternary Search
- [x] C
- [x] C++
- [x] C# - @p-soham11
- [x] CoffeeScript
- [ ] Dart - @heisastark
- [ ] Go - @Ratna04priya
- [x] Java
- [x] JS
- [ ] Kotlin - @saloniankita23
- [x] Php - @BhardwajShweta25
- [x] Python
- [x] Readme - @shwetagurnani
- [ ] Ruby - @atarax665
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by atarax665**
Fixes #2176 

#### Checklist:
- [ ] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
Adding ternary search in Ruby

#### Languages Used:
Ruby

#### Files Added:
Ternary_Search.rb

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by atarax665** [comment]
@jainaman224 Please review the PR

File: Ternary_Search/Ternary_Search.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+def ternarySearch(l, r, key, ar) #function to perform ternary search
+
```
Comment (asha15) [inline Ternary_Search/Ternary_Search.rb:2]: remove this line

File: Ternary_Search/Ternary_Search.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    if (r >= l)
+        #find mid1 and mid2
+        mid1 = l + (r - l) / 3
+        mid2 = r - (r - l) / 3
```
Comment (asha15) [inline Ternary_Search/Ternary_Search.rb:6]: keep a line space after this

File: Ternary_Search/Ternary_Search.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        #then repeat the Search operation in that region 
+        elsif key < ar[mid1] 
+            ternarySearch(l, mid1 - 1, key, ar)
+        elsif (key > ar[mid2]) 
+            ternarySearch(mid2 + 1, r, key, ar)
+        else 
+            ternarySearch(mid1 + 1, mid2 - 1, key, ar) 
+        end
+    end
+end
```
Comment (asha15) [inline Ternary_Search/Ternary_Search.rb:21]: keep a line space after this

File: Ternary_Search/Ternary_Search.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+n= gets.to_i #length of array
```
Comment (asha15) [inline Ternary_Search/Ternary_Search.rb:22]: keep space around =

File: Ternary_Search/Ternary_Search.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+arr = []
+while n>0
+  num = gets.chomp.to_i
+  arr.push(num)
+  n = n-1
```
Comment (asha15) [inline Ternary_Search/Ternary_Search.rb:27]: correct the indentation 
please keep space around operators

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by atarax665** [comment]
No, all the issues have been resolved.

**Comment by somya-kapoor** [comment]
please squash your commits

**Comment by atarax665** [comment]
@somya-kapoor closing this pr because too many merges were making the squash process complicated. Started a new PR tho.

**Pull request closed without merging**